sound/soc/sof/intel/probe.h

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: (GPL-2.0 OR BSD-3-Clause) */
+/*
+ * This file is provided under a dual BSD/GPLv2 license.  When using or
+ * redistributing this file, you may do so under either license.
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Cezary Rojewski <cezary.rojewski@intel.com>
+ */
+
+#ifndef __SOF_INTEL_PROBE_H
+#define __SOF_INTEL_PROBE_H
+
+#include <sound/sof/header.h>
+
+struct snd_sof_dev;
+
+#define INVALID_NODE_ID UINT_MAX
+
+struct sof_probe_dma {
+	unsigned int stream_tag;
+	unsigned int dma_buffer_size;
+} __packed;
+
+enum sof_probe_type {
+	SOF_PROBE_TYPE_INPUT = 0,
+	SOF_PROBE_TYPE_OUTPUT,
+	SOF_PROBE_TYPE_INTERNAL
+};
+
+enum sof_connection_purpose {
+	SOF_CONNECTION_PURPOSE_EXTRACT = 0,
+	SOF_CONNECTION_PURPOSE_INJECT,
+	SOF_CONNECTION_PURPOSE_INJECT_REEXTRACT,
+};
+
+struct sof_probe_point_desc {
+	unsigned int buffer_id;
+	enum sof_connection_purpose purpose __aligned(4);
```

Comment:
is this intended to be use for IPC? If yes, please remove the enum. we don't have enums in any IPC definition.

Reply:
Ack.